src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,76 @@
+import classNames from 'classnames';
+import { SearchLink } from './SearchLink';
+import { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../utils/searchHelper';
+
+const centuryOptions = ['16', '17', '18', '19', '20'];
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const filteredByQuery = searchParams.get('query') || '';
+  const filteredBySex = searchParams.get('sex') || '';
+  const filteredByCenturies = searchParams.getAll('centuries');
+  const sortedBy = searchParams.get('sortedBy') || '';
+
+  const handleChangingQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newQuery = event.target.value.trim() ? event.target.value : null;
+    const newSearchParam = getSearchWith(searchParams, { query: newQuery });
+
+    setSearchParams(newSearchParam);
+  };
+
+  const handleTogglingCenturies = (addedCentury: string) => {
+    return {
+      centuries: filteredByCenturies.includes(addedCentury)
+        ? [...filteredByCenturies].filter(century => century !== addedCentury)
```

Comment:
```suggestion
        ? filteredByCenturies.filter(century => century !== addedCentury)
```
it's redundant to make copy here, cause the filter method doesn't mutate the array